bin/oneoff/import_unit_details.rb

```diff
@@ -191,6 +220,72 @@ def get_validated_lesson_pairs(script, chapters)
   validated_lesson_pairs
 end
 
+# Validates the following preconditions:
+
+# 1. Every CB chapter has a corresponding LessonGroup in Code Studio.
+# 2. If the CB unit uses chapters, then for each Code Studio chapter containing
```

Comment:
Code Studio Lesson Group?

Reply:
This second condition is something I'm struggling to wrap my head around for some reason

Reply:
the thing I am doing is a bit complicated and I am concerned that I may be overlooking a much simpler way. let's look at csd1-2020:
![Screen Shot 2020-11-11 at 12 42 34 PM](https://user-images.githubusercontent.com/8001765/98862253-70326580-241b-11eb-8a20-0acba66d4845.png)

The things I need to do are:

1. ignore any lessons which do not have lesson plans. in this case, those are the 2 lockable lessons visible in the screenshot.

2. ignore any lesson groups which, having ignored lessons without lesson plans, are now empty

This is a different order from what I have implemented. does this order sound easier to follow? 

I think it sounds clearer to me, but the problem I ran into was that I don't actually want to modify the lesson groups, so it is a bit easier to implement swap the order and do

1. ignore lesson groups which will become empty after ignoring their lessons without lesson plans, and then 

2. (inside the smaller loop) for each lesson plan which I'm not ignoring, to ignore the lessons without lesson plans. 

having written that, the way I'm doing it sounds a lot harder to explain. I'll try to think of a way to simplify this but please let me know if you have any ideas!

Reply:
If I do things in this order:
1. ignore lessons without lesson plans
2. ignore any empty chapters

then requirement 2 just says that any Code Studio lesson group which hasn't been ignored by steps 1 and 2 should have a corresponding chapter on CB.

Reply:
Ahhh ok I see what you are saying. I don't know if its worth re-implementing if its working. Maybe just point to this PR discussion in the comments so someone can find their way back here?

Reply:
I changed the comments to match the order of how i was doing things in the code, and tried to explain it a bit better.